ci: Move codespell into GitHub action workflow, apply codespell version.

projectgus opened the pull request:
* Similar to ruff.yaml, it's simpler to run the `codespell` command directly from a workflow file.
* Applies a specific version to codespell, same as pre-commit (introduced in #12157)

~~Additional, optional commit in this branch:~~

EDIT: Dropped this commit as `yq` is only packaged in Ubuntu starting from "lunar", 23.04.

* ~~Read the CI tool versions (ruff, codespell) from `.pre-commit-config.yaml`. This means there is now a single canonical source for these versions, at the cost of some ugly "yq" YAML parsing in the workflow steps.~~

~~Can drop the second commit if it's too fiddly, and~~ will add comments instead to remind anyone updating that the version is copied in two places.

*This work was funded through GitHub Sponsors.*

projectgus committed:
github/workflows: Move codespell to a GitHub workflow, version it.

Similar to ruff.yaml, it's simpler to run the codespell command directly
from a workflow file.  And developers can run codespell directly from the
command line without the need for options, or just use pre-commit.

This commit also applies a specific version to codespell, same as
pre-commit (introduced in a166d805f43).

This work was funded through GitHub Sponsors.

Signed-off-by: Angus Gratton <angus@redyak.com.au>

projectgus committed:
github/workflows: Add comments where tool versions need to be in sync.

This work was funded through GitHub Sponsors.

Signed-off-by: Angus Gratton <angus@redyak.com.au>

projectgus commented:
@Josverl This overlaps with some of the (very useful) tooling & CI improvements you've been making, so PTAL.

dpgeorge commented on .github/workflows/codespell.yml:
```diff
@@ -0,0 +1,12 @@
+# https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
```
Why is this comment needed?  Other workflow files don't have it.

projectgus replied:
I copied it from ruff.yml which has it, but actually both these workflow files don't use much of what's documented there.

Will remove from both.

dpgeorge commented on .github/workflows/codespell.yml:
```diff
@@ -0,0 +1,12 @@
+# https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
+name: Check spelling with codespell
+on: [push, pull_request]
+jobs:
```
Please put blank lines around the top-level sections, to be consistent with, eg, `code_formatting.yml`.

projectgus replied:
As above, this file is a copy of ruff.yml. I'll change them both.

dpgeorge replied:
Oh, I see.  Yeah, would be good to change both.

projectgus committed:
github/workflows: Standardise formatting of ruff.yml.

All the other workflow YAML files use vertical whitespace around top-level
items.

Also remove spurious comment, the features in the linked doc aren't
actually used in this workflow (any more?).

This work was funded through GitHub Sponsors.

Signed-off-by: Angus Gratton <angus@redyak.com.au>
